pennylane/transforms/qcut.py

```diff
@@ -828,6 +828,9 @@ def _reshape_results(results: Sequence, shots: int) -> List[List]:
     is determined by the number of shots and whose number of columns is determined by the number of
     cuts.
     """
+    if qml.active_return():
+        results = [qml.math.stack(r) if isinstance(r, tuple) else r for r in results]
```

Comment:
When is `if isinstance(r, tuple)` `True`? I.e., what results are we stacking and why?

Choosing a more descriptive name for `r` would be great.

Reply:
I don't know the inner workings of qcut, but each generated tape only contains a number of expval measurements (in the exact case) or a number of sample measurements (in the Monte Carlo case). So stacking the results won't create any ragged arrays. We do the stacking so that previous functionality is maintained.

Sure, I can rename `r` to be more descriptive.